includes/classes/SupportMonitor/DBQueryMonitor.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Database Queries Monitor. A submodule of Support Monitor to report heavy SQL queries executed on staging.
+ *
+ * @since  x.x
+ * @package 10up-experience
+ */
+
+namespace TenUpExperience\SupportMonitor;
+
+/**
+ * DBQueryMonitor class
+ */
+class DBQueryMonitor {
+
+	/**
+	 * The transient name
+	 *
+	 * This transient stores all the queries logged.
+	 */
+	const TRANSIENT_NAME = 'tenup_experience_db_queries';
+
+	/**
+	 * Setup module
+	 */
+	public function setup() {
+		$production_environment = Monitor::instance()->get_setting( 'production_environment' );
+		if ( 'no' === $production_environment || apply_filters( 'tenup_experience_log_heavy_queries', false ) ) {
+			add_filter( 'query', [ $this, 'maybe_log_query' ] );
+		}
+	}
+
+	/**
+	 * Conditionally log a query
+	 *
+	 * Get all potential heavy queries (CREATE, ALTER, etc.) and store it,
+	 * ignoring transients and options by default.
+	 *
+	 * @param string $query The SQL query.
+	 * @return string
+	 */
+	public function maybe_log_query( $query ) {
+		global $wpdb;
+
+		if ( ! preg_match( '/^\s*(create|alter|truncate|drop|insert|delete|update|replace)\s/i', $query ) ) {
```

Comment:
@felipeelia I'm not sure if this is intentional, but I want to point out that logging DELETE or UPDATE statements are likely going to fill the database if you figure in the case we are running a migration or just some heavy maintenance is being done on a non-production environment (like preprod). I also question the value of having that information logged since it's so common. Maybe there's a way to identify whether it's not a normal WP action that was initiated from the backend?

Reply:
@bengreeley yeah, that is still being worked out. The plan right now is to log only insert/update/delete queries with a nested select or queries that are really big. I'll work on it during the next few days. Thanks!

Reply:
this has been added to the code already ^